src/com/cloudogu/ces/cesbuildlib/Git.groovy

```diff
@@ -260,4 +325,88 @@ class Git implements Serializable {
             script.sh "git ${args}"
         }
     }
-}
+
+    /**
+     * Finishes a git flow release and pushes all merged branches to remote
+     *
+     * Only execute this function if you are already on a release branch
+     *
+     * @param releaseVersion the version that is going to be released
+     */
+    void finishGitRelease(String releaseVersion) {
+        String branchName = getBranchName()
+
+        // Check if tag already exists
+        if (tagExists("${releaseVersion}")){
+            error("You cannot build this version, because it already exists.")
+        }
+
+        // Make sure all branches are fetched
+        script.sh "git config 'remote.origin.fetch' '+refs/heads/*:refs/remotes/origin/*'"
+        executeGitWithCredentials("fetch --all")
+
+        // Make sure there are no changes on develop
+        if (developHasChanged(branchName)){
+            error("There are changes on develop branch that are not merged into release. Please merge and restart process.")
+        }
+
+        // Make sure any branch we need exists locally
+        script.sh "git checkout ${branchName}"
+        executeGitWithCredentials("pull origin ${branchName}")
+        script.sh "git checkout develop"
+        executeGitWithCredentials("pull origin develop")
+        script.sh "git checkout master"
+        executeGitWithCredentials("pull origin master")
+
+        // Merge release branch into master
+        script.sh "git merge --no-ff ${branchName}"
+
+        // Create tag. Use -f because the created tag will persist when build has failed.
+        executeGitWithCredentials("tag -f -m 'release version ${releaseVersion}' ${releaseVersion}")
+
+        // Merge release branch into develop
+        script.sh "git checkout develop"
+        script.sh "git merge --no-ff ${branchName}"
+
+        // Delete release branch
+        script.sh "git branch -d ${branchName}"
```

Comment:
Please introduce and use `branch()` method. Not repeating `script.sh` all over will improve readability

Reply:
What should this `branch()` method do?
Anyway i removed all of the `script.sh` calls and replaced them by calls to the git object.